app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/local/eventDao/eventDaoImpls/EventDaoFakeImpl.kt

```diff
@@ -0,0 +1,289 @@
+
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.local.entities.EventEntity
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.local.eventDao.IEventDao
+import com.realityexpander.tasky.agenda_feature.util.EventId
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.time.ZonedDateTime
+import javax.inject.Inject
+
+// Simulates a local database
+
+class EventDaoFakeImpl @Inject constructor(): IEventDao {
+
+    // • CREATE
+
+    override suspend fun createEvent(event: EventEntity) {
+        createEventInFakeDatabase(event)
+    }
+
+
+    // • READ
+
+    override suspend fun getEventsForDay(zonedDateTime: ZonedDateTime): List<EventEntity> {
+        return getEventsForDayInFakeDatabase(zonedDateTime)
+    }
+
+    override suspend fun getAllEventsFlow(): Flow<List<EventEntity>> {
+        return getAllEventsFlowInFakeDatabase()
+    }
+
+    override suspend fun getEventById(eventId: EventId): EventEntity? {
+        return getEventByIdInFakeDatabase(eventId)
+    }
+
+    override suspend fun getAllEvents(): List<EventEntity> {
+        return getAllEventsInFakeDatabase()
+    }
+
+
+    // • UPDATE
+
+    override suspend fun updateEvent(event: EventEntity): Boolean {
+        return try {
+            updateEventInFakeDatabase(event)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    // • DELETE
+
+    // Only marks the event as deleted
+    override suspend fun deleteEventById(eventId: EventId): Boolean {
+        return try {
+            deleteEventByIdInFakeDatabase(eventId)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    override suspend fun deleteFinallyByEventIds(eventIds: List<EventId>): Boolean {
+        return try {
+            deleteFinallyByEventIdsInFakeDatabase(eventIds)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    override suspend fun getDeletedEventIds(): List<EventId> {
+        return getDeletedEventIdsInFakeDatabase()
+    }
+
+    override suspend fun clearAllEvents(): Boolean {
+        return try {
+            clearAllEventsInFakeDatabase()
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    ////////////////////////////////// FAKE DATABASE //////////////////////////////////
+
+    private val events = mutableListOf<EventEntity>()
+    private val eventsFlow = flow {
+        var lastEvents= emptyList<EventEntity>()
+        while(true) {
+            if(!lastEvents.containsAll(events)) {
+                lastEvents = events.toList()
+                emit(events)
+            }
+            delay(100)
+        }
+    }
```

Comment:
You could also just use a StateFlow here, then you're immediately notified about changes